README.md

```diff
@@ -1,33 +1,53 @@
+|`clientSecret`|source|The Client Secret for Marketo Instance|true|NONE|ZZZv0Mev29vNm5vIyMwTa43lioVoBT7N|
+|`clientEndpoint`|source|The Endpoint for Marketo Instance|true|NONE| https://\<instance\>.mktorest.com |
+|`pollingPeriod`|source|Polling time for CDC mode. Less than 10s is not recommended |false|`1m`| `10s`, `1m`, `5m`, `10m`, `30m`, `1h` |
+|`fields`|source|comma seperated fields to fetch from Marketo Leads|false|`id, createdAt, updatedAt, firstName, lastName, email`| `company, jobTitle, phone, personSource` etc... |
+
+>Note: By default **`id, createdAt, updatedAt`** is prepended to `fields` config. So no need to add that explictly. For eg: if you want to request `email, company, phone` fields, then it will be requested as **`id, createdAt, updatedAt, email, company, phone`**
+
+## Source
+Marketo source connector connects to Marketo instance through the REST API with provided configuration, using `clientID` and `clientSecret`. Once connector is started `Configure` method is called to parse configurations and validate them. After that `Open` method is called to establish connection to Marketo instance with provided position. Once connection is established `Read` method is called which calls current iterator's `Next` method to fetch next record. `Teardown` is called when connector is stopped. 
+
+### Snapshot Iterator
+Snapshot iterator is used first to extract bulk data from Marketo instance. [Bulk Lead Extract API](https://developers.marketo.com/rest-api/bulk-extract/bulk-lead-extract/) is used with `createdAt` filter which permits datetime ranges up to 31 days, so we will need to run multiple jobs and combine the results. In order to get started we need to find the oldest lead created in the instance. To know the date [querry all folders](https://developers.marketo.com/rest-api/assets/folders/#browse) with maxdepth of 1 which will give us a list of all the top-level folders in the instance. Then collecting `createdAt` dates, parse them, and find the oldest date. This method works because some default, top-level folders are created with the instance and no leads could be created before then. `fields` from config also requested along with `createdAt` filter. To find the available fields for your target instance using the [Describe Lead 2 endpoint](https://developers.marketo.com/rest-api/endpoint-reference/lead-database-endpoint-reference/#!/Leads/describeUsingGET_6) which return an exhaustive list including both standard and custom fields.
+
+**Exporting Job involves 4 APIS**
+- Create a Job -> `/bulk/v1/leads/export/create.json`
+- Enqueue a Job ->`/bulk/v1/leads/export/{{exportID}}/enqueue.json`
+- Wait for Job to Complete -> `/bulk/v1/leads/export/{{exportID}}/status.json`
+- Get Your Leads -> `/bulk/v1/leads/export/{{exportID}}/file.json`
+
+After cycling through every 31-day range from the first lead until now, we’ll have a complete set. Then flush bulk data to the buffer which pushes data one by one to Conduit.
```

Comment:
So we first cycle through all 31-day ranges, and then send data to Conduit?

Reply:
Yes. After each cycle we'll send the records, then we go to next cycle. We can wait till all cycles to complete, but that may take minutes to hours for completion. Also suppose we cancel the pipeline no need to cancel all export jobs(running cycles). Canceling current job will stop the pipeline(applicable only to snapshot mode)

Reply:
> So we first cycle through all 31-day ranges, and then send data to Conduit?

What I meant is: do we cycle through **ALL** ranges first, and only once we are done fetching all ranges, we then send records to Conduit.

> Yes. After each cycle we'll send the records, then we go to next cycle.

OK, so we fetch a range, send data to Conduit, we fetch another, send data to Conduit, and so on? 

The text in ReadMe says: 

> After cycling through every 31-day range from the first lead until now, we’ll have a complete set. Then flush bulk data to the buffer which pushes data one by one to Conduit.

Here it says that once we have a complete set (all ranges), we then send data to Conduit.

Reply:
Readme file updated